src/main/java/org/broadinstitute/dsm/route/FilterRoute.java

```diff
@@ -348,14 +343,38 @@ public static void addParticipantDataFilters(Map<String, String> queryConditions
         }
     }
 
-    public static void addParticipantDataCondition(StringBuilder newCondition, boolean first, ParticipantDataDto participantData) {
-        if (first) {
+    public static void addConditionForRange(Filter filter, String tmpName, StringBuilder newCondition,
+                                            AtomicBoolean participantAdded, AtomicBoolean first, ParticipantDataDto participantData,
+                                            JsonObject dataJsonObject) {
+        Object value1 = filter.getFilter1().getValue();
+        Object value2 = filter.getFilter2().getValue();
+        if (value1 == null || value2 == null) {
+            return;
+        }
+        boolean inNumberRange = value1 instanceof Double && value2 instanceof Double && dataJsonObject.get(tmpName) != null
+                && Double.compare(dataJsonObject.get(tmpName).getAsDouble(), (Double) value1) > 0
+                && Double.compare(dataJsonObject.get(tmpName).getAsDouble(), (Double) value2) < 0;
```

Comment:
would it still be inNumberRange if value1 = 5 and value2 = 3

Reply:
Didn't know that I had to consider it from business logic perspective. I'll add to check that